src/App.tsx

```diff
@@ -10,51 +12,150 @@ export function App() {
   const [activeTabIndex, setActiveTabIndex] = useState(0);
   const [platform, setPlatform] = useState<browser.runtime.PlatformOs>("mac");
   const [search, setSearch] = useState<string>("");
-  const [tabs, setTabs] = useState<browser.tabs.Tab[]>([]);
+  const [tabs, setTabs] = useState<SearchResult[]>([]);
+  const [currentSearchEngine, setCurrentSearchEngine] = useState<
+    string | undefined
+  >(undefined);
 
   const resultsListref = useRef<HTMLDivElement>(null);
+  const searchInputRef = useRef<HTMLInputElement>(null);
 
   const debouncedTerm = useDebounce(search, 150);
   const arrowDownPressed = useKeyPress("ArrowDown");
   const arrowUpPressed = useKeyPress("ArrowUp");
 
-  const getInitialTabs = async () => {
-    const tabs = await browser.tabs.query({});
-    setTabs(tabs);
+  const getAllBookmarks = async (): Promise<
+    browser.bookmarks.BookmarkTreeNode[]
+  > => {
+    const bookmarkTree = await browser.bookmarks.getTree();
+    const bookmarks: browser.bookmarks.BookmarkTreeNode[] = [];
+
+    const traverse = (nodes: browser.bookmarks.BookmarkTreeNode[]) => {
```

Comment:
added as a comment but I think we can replace it all with `browser.bookmarks.search`